File: inst_san_martin_b/services/dni.go
package services

import (
	"regexp"
	"strings"
)

var dniArgentino7u8Digitos = regexp.MustCompile(`^\d{7,8}$`)

/* ValidateDNIAr exige 7 u 8 digitos numericos solamente (sin puntos ni letras). */
func ValidateDNIAr(dni string) (msg string, ok bool) {
	d := strings.TrimSpace(dni)
	if d == "" {
		return "El DNI es obligatorio", false
	}
	if !dniArgentino7u8Digitos.MatchString(d) {
		return "El DNI debe tener 7 u 8 digitos numericos (sin puntos ni letras)", false
	}
	return "", true
}

File: inst_san_martin_f/src/utils/dni.js
/** DNI: exactamente 7 u 8 caracteres, solo dígitos (sin puntos ni letras). */
export const validateDNI = (dni) => {
  const d = dni != null ? String(dni).trim() : '';
  if (!d) {
    return { ok: false, message: 'El DNI es obligatorio' };
  }
  if (!/^\d{7,8}$/.test(d)) {
    return {
      ok: false,
      message: 'El DNI debe tener 7 u 8 dígitos numéricos (sin puntos ni letras)'
    };
  }
  return { ok: true, message: '' };
};
